specification.md

```diff
@@ -595,6 +595,16 @@ customary in the language in which the library is implemented
 #### Methods
 
 
+<a name="jobexecutor-constructor"></a>
+```java
+JobExecutor(String url)
+```
+
+Instantiates an `JobExecutor` object and binds it to the backend specified by
+URL `url`.  The URL will also encode any authentication details needed to
```

Comment:
Could this constructor throw an exception if it fails to connect, either due to authentication or due to a non-responsive/missing/dead executor?

Reply:
Yes, that makes sense of course - will add.

Reply:
I think we need to hit the breaks a bit since things seem to be getting a bit out of control. Remote connections are beyond the scope of what we are doing. If we are talking about them, something has gone wrong.

Reply:
See note in #23 - I don't think URL needs to mean remote.  Feel free to replace `URL` with `endpoint specifier` or whatever.

Reply:
Yeah, I interpreted this to be an implementation/executor-specific URL that doesn't necessarily need to be remote.  In the case of Flux, I expect it to be the [URI that you would normally pass to `flux_open`](https://flux-framework.readthedocs.io/projects/flux-core/en/latest/man3/flux_open.html?highlight=flux_open#description) (e.g., `file:///tmp/flux-ndacld8763/local/0`).  Useful for when you have a bunch of Flux instances running at once.  In Slurm's case, maybe it is the location of your `slurm.conf` (assuming you have it in a non-default location or multiple slurm instances running on your cluster for whatever reason).

Side benefit that in Flux's case (and I'm sure others), it could easily be adapted to the remote case (e.g., `ssh://stephen@mycluster/path/to/domain-socket`).

Reply:
PS - I just saw the discussion over in #23, and my comment above doesn't add much to what was already discussed there.  So don't worry about rebutting my comment here.

Reply:
Added the exception.